databoard/testing.py

```diff
@@ -0,0 +1,120 @@
+
+
+def create_test_db():
+    """Create an empty test db and setup the files s.
+
+    The different settings should be set as environment variables. The
+    environment variables to set are:
+
+    * DATABOARD_STAGE: stage of the database. Need to be `'TESTING'` to use
+      this function;
+    * DATABOARD_USER: database login;
+    * DATABOARD_PASSWORD: database password;
+    * DATABOARD_DB_URL_TEST: database URL.
+    """
+    if os.getenv('DATABOARD_STAGE') in ['TEST', 'TESTING']:
+        shutil.rmtree(deployment_path, ignore_errors=True)
+        os.makedirs(deployment_path)
+        # TODO: not sure it is necessary to have the fabfile copied for local
+        # test
+        shutil.copyfile(os.path.abspath('fabfile.py'),
+                        os.path.join(deployment_path, 'fabfile.py'))
+        os.makedirs(ramp_config['ramp_kits_path'])
+        os.makedirs(ramp_config['ramp_data_path'])
+        os.makedirs(ramp_config['ramp_submissions_path'])
+        # create the empty database
+        db.session.close()
+        db.drop_all()
+        db.create_all()
+        setup_files_extension_type()
+    else:
+        raise AttributeError('DATABOARD_STAGE should be set to TESTING for '
+                             '`deploy` to work')
+
+
+def add_users():
+    create_user(
+        name='test_user', password='test',
+        lastname='Test', firstname='User',
+        email='test.user@gmail.com', access_level='asked')
+    approve_user('test_user')
+    create_user(
+        name='test_iris_admin', password='test',
+        lastname='Admin', firstname='Iris',
+        email='iris.admin@gmail.com', access_level='user')
+
+
+def _setup_ramp_kits_ramp_data(problem_name):
+    # TODO: This function does not have a unit test but only used in
+    # integration testing.
+    problem_kits_path = os.path.join(ramp_config['ramp_kits_path'],
+                                     problem_name)
+    ramp_kits_url = 'https://github.com/ramp-kits/{}.git'.format(problem_name)
+    ramp_kits_repo = Repo.clone_from(ramp_kits_url, problem_kits_path)
+
+    problem_data_path = os.path.join(ramp_config['ramp_data_path'],
+                                     problem_name)
+    ramp_data_url = 'https://github.com/ramp-data/{}.git'.format(problem_name)
+    ramp_data_repo = Repo.clone_from(ramp_data_url, problem_data_path)
+
+    current_directory = os.getcwd()
+    os.chdir(problem_data_path)
+    subprocess.check_output(["python", "prepare_data.py"])
+    os.chdir(problem_kits_path)
+    subprocess.check_output(["jupyter", "nbconvert", "--to", "html",
+                             "{}_starting_kit.ipynb".format(problem_name)])
+    os.chdir(current_directory)
+
+
+def add_problems():
+    problems = ['iris', 'boston_housing']
+    for problem_name in problems:
+        _setup_ramp_kits_ramp_data(problem_name)
+        add_problem(problem_name)
+
+
+# def _add_problem_and_event(problem_name, test_user_name):
```

Comment:
is this to be removed?

Reply:
I did not write yet the `add_event` yet.